core/org.wso2.carbon.user.core/src/main/java/org/wso2/carbon/user/core/claim/DefaultClaimManager.java

```diff
@@ -86,10 +86,14 @@ public String getAttributeName(String claimURI) throws UserStoreException {
         if (claimCache.isInvalid()) {
             this.claimMapping = getClaimMapFromDB();
         }
-        ClaimMapping mapping = claimMapping.get(claimURI);
-        if (mapping != null) {
-            return mapping.getMappedAttribute();
+
+        if (claimURI != null) {
+            ClaimMapping mapping = claimMapping.get(claimURI);
+            if (mapping != null) {
+                return mapping.getMappedAttribute();
+            }
         }
+
         return null;
```

Comment:
Yes. I agree with your point. But returning "NULL" is the behavior we had already and there can be other dependent components which have implemented based on this behavior. So changing this would affect those components as well.